AirCasting/Utils/Bluetooth/BluetoothManager.swift

```diff
@@ -376,18 +376,19 @@ final class BluetoothManager: NSObject, BluetoothCommunicator, CBCentralManagerD
     
     func peripheral(_ peripheral: CBPeripheral, didUpdateNotificationStateFor characteristic: CBCharacteristic, error: Error?) {
         if error != nil {
-            Log.error("[SD Sync] \(error)")
+            Log.error("[SD Sync] \(String(describing: error))")
         }
     }
     
     func peripheral(_ peripheral: CBPeripheral, didUpdateValueFor characteristic: CBCharacteristic, error: Error?) {
+        let value = characteristic.value
         characteristicsMappingLock.lock()
         defer { characteristicsMappingLock.unlock()}
         guard let containgObserver = charactieristicsMapping.first(where: { CBUUID(string: $0.key.value) == characteristic.uuid }) else { return }
         containgObserver.value.forEach { observer in
             observer.triggerCounter += 1
             guard error == nil else { observer.action(.failure(error!)); return }
-            callbackQueue.async { observer.action(.success(characteristic.value)) }
+            callbackQueue.async { observer.action(.success(value)) }
```

Comment:
Due to a race condition, sometimes characteristic.value was changing during this function's execution and we were loosing the data.